src/Sparrow/Json/JsonContextPoolBase.cs

```diff
@@ -167,10 +246,27 @@ private void CleanThreadLocalState()
             {
                 current = MaybeGetCurrentContextStack()?.Head;
 
+                _contextStacksByThreadId.TryRemove(NativeMemory.CurrentThreadStats.InternalId, out _);
+
+                if (_releaser != null)
+                {
+                    // we want to clear our JsonContextPool's current thread's state from the _releaser, but to avoid touching any other states
+                    foreach (var threadIdHolder in _threadIds)
+                    {
+                        if (Interlocked.CompareExchange(ref threadIdHolder.ThreadId, -1, NativeMemory.CurrentThreadStats.InternalId) == NativeMemory.CurrentThreadStats.InternalId)
```

Comment:
I don't think this is a  good idea. How do we know that this is from _our_ context pool and not from another?

Reply:
Probably @maximburyak should confirm, but `_threadIds` are local to the context pool and we are no longer hijacking the ThreadIdHolders from other pools, we are always creating an instance for this specific pool in `EnsureCurrentThreadContextWillBeReleased` or reuse one from the ones that were previosly created by that pool.

Reply:
Yes, you are correct.